### PR TITLE
Improve instance filtering and listing

### PR DESCRIPTION
**Pull request by marcesher**

### CODE REVIEW

File: scripts/ec2/ls.coffee [hunk 1]
```diff
@@ -29,29 +30,36 @@ DAYS_CONSIDERED_SOON = 2
 ec2 = require('../../ec2.coffee')
 
 getArgParams = (arg, filter = "all", opt_arg = "") ->
-  instances = []
+  filterValues = []
```
Comment (marcesher) [inline scripts/ec2/ls.coffee:33]: because `instances` is a flat-out wrong variable name

File: scripts/ec2/ls.coffee [hunk 2]
```diff
@@ -29,29 +30,36 @@ DAYS_CONSIDERED_SOON = 2
...
   if filter == "mine"
     params['Filters'] = [{Name: 'tag:Creator', Values: [opt_arg]}]
   else if filter == "chat"
     params['Filters'] = [{Name: 'tag:CreatedByApplication', Values: [filter]}]
-  else if filter == "filter" and instances.length
-    params['Filters'] = [{Name: 'tag:Name', Values: ["*#{instances[0]}*"]}]
-  else if instances.length
-    params['InstanceIds'] = instances
+  else if filter == "filter" and filterValues.length
+    params['Filters'] = [{Name: 'tag-value', Values: ["*#{filterValues[0]}*"]}]
```
Comment (marcesher) [inline scripts/ec2/ls.coffee:49]: using `tag-value` will cause it to apply the search filter across any tags for instances. Probably the single most important change in this PR

File: scripts/ec2/ls.coffee [hunk 3]
```diff
@@ -152,27 +160,30 @@ messages_from_ec2_instances = (instances) ->
     name = '[NoName]'
     for tag in instance.Tags when tag.Key is 'Name'
       name = tag.Value
+    description = ''
+    for tag in instance.Tags when tag.Key is 'Description'
+      description = tag.Value
 
     messages.push({
-      time: moment(instance.LaunchTime).format('YYYY-MM-DD HH:mm:ssZ')
+      time: moment(instance.LaunchTime).format('YYYY-MM-DD')
```
Comment (marcesher) [inline scripts/ec2/ls.coffee:168]: Time will not be useful for most users; removing in favor of giving more space for description

File: scripts/ec2/ls.coffee [hunk 4]
```diff
@@ -152,27 +160,30 @@ messages_from_ec2_instances = (instances) ->
...
       state: instance.State.Name
       id: instance.InstanceId
-      image: instance.ImageId
```
Comment (marcesher) [inline scripts/ec2/ls.coffee:160]: only useful for power users, who will have console access

File: scripts/ec2/ls.coffee [hunk 5]
```diff
@@ -152,27 +160,30 @@ messages_from_ec2_instances = (instances) ->
...
-      az: instance.Placement.AvailabilityZone
-      subnet: instance.SubnetId
       type: instance.InstanceType
       ip: instance.PrivateIpAddress
       name: name || '[NoName]'
+      description: description || ''
```
Comment (marcesher) [inline scripts/ec2/ls.coffee:174]: 2nd most important addition with this PR.

**Comment by OrlandoSoto** [comment]
thanks @marcesher !

**Pull request merged**